lib/experimental/fonts/font-library/class-wp-rest-font-collections-controller.php

```diff
@@ -68,13 +71,61 @@ public function register_routes() {
 	 *
 	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
 	 */
-	public function get_items( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		$collections = array();
-		foreach ( WP_Font_Library::get_font_collections() as $collection ) {
-			$collections[] = $collection->get_config();
+	public function get_items( $request ) {
+		$collections_all = WP_Font_Library::get_font_collections();
+
+		$page        = $request['page'];
+		$per_page    = $request['per_page'];
+		$total_items = count( $collections_all );
+		$max_pages   = ceil( $total_items / $per_page );
+
+		if ( $page > $max_pages && $total_items > 0 ) {
+			return new WP_Error(
+				'rest_post_invalid_page_number',
+				__( 'The page number requested is larger than the number of pages available.', 'gutenberg' ),
```

Comment:
Since this is exactly the same string as in Core, I think we can use the `default` text domain so it doesn't have to be re-translated.
```suggestion
				__( 'The page number requested is larger than the number of pages available.', 'default' ),
```